File: opentsdb-consumer/src/main/java/net/opentsdb/kafka/consumer/netty/TsdbClientHandler.java
package net.opentsdb.kafka.consumer.netty;

import org.jboss.netty.channel.*;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class TsdbClientHandler extends SimpleChannelUpstreamHandler {
  private static final Logger logger = LoggerFactory.getLogger(TsdbClientHandler.class);

  @Override
  public void handleUpstream(ChannelHandlerContext ctx, ChannelEvent e) throws Exception {
    if (e instanceof ChannelStateEvent) {
      logger.info(e.toString());
    }

    super.handleUpstream(ctx, e);
  }

  @Override
  public void messageReceived(ChannelHandlerContext ctx, MessageEvent e) {
    logger.error(e.getMessage().toString());
  }

  @Override
  public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) {
    logger.warn("Unexpected exception from downstream: {}", e.getCause());
    e.getChannel().close();
  }
}

File: opentsdb-client/src/main/scala/net/opentsdb/client/netty/TsdbClientHandler.scala
package net.opentsdb.client.netty

import org.jboss.netty.channel._
import org.slf4j.{LoggerFactory, Logger}

class TsdbClientHandler extends SimpleChannelUpstreamHandler {
  private final val logger: Logger = LoggerFactory.getLogger(classOf[TsdbClient])



  override def handleUpstream(ctx: ChannelHandlerContext, e: ChannelEvent) = {
    e match {
      case e: ChannelStateEvent => logger.info(e.toString)
      case _ =>
    }

    super.handleUpstream(ctx, e)
  }

  override def messageReceived(ctx: ChannelHandlerContext, e: MessageEvent) = {
    logger.error(e.getMessage.toString)
  }

  override def exceptionCaught(ctx: ChannelHandlerContext, e: ExceptionEvent) = {
    logger.warn("Unexpected exception from downstream: {}", e.getCause)
    e.getChannel.close()
  }
}
